src/threadpool_scheduler.cpp

```diff
@@ -301,6 +301,9 @@ static void recursive_spawn_worker_thread(threadpool_data* impl, std::size_t ind
 
 } // namespace detail
 
+threadpool_scheduler::threadpool_scheduler(threadpool_scheduler&& other)
+        : impl(std::move(other.impl)) {}
+
```

Comment:
Put this in the header file as an inline function.

Reply:
it will break due to the forward declaration of `detail::threadpool_data`